doc/user/content/transform-data/patterns/partition-by.md

```diff
@@ -0,0 +1,143 @@
+---
+title: "PARTITION BY"
+description: "Specifying the sort order of the data."
+aliases:
+  - /guides/partition-by/
+  - /sql/patterns/partition-by/
+menu:
+  main:
+    parent: 'sql-patterns'
+---
+
+A few types of Materialize collections are durably written to storage: [materialized views](/sql/create-materialized-view/), [tables](/sql/create-table), and [sources](/sql/create-source).
+
+Internally, Materialize stores these durable collections in an [LSM-tree](https://en.wikipedia.org/wiki/Log-structured_merge-tree)-like structure. Each collection is made up of a set of
+**runs** of data, each run is sorted and then split up into individual **parts**, and those parts are written to object storage and retrieved only when necessary to satisfy a query. Materialize will also periodically **compact** the data it stores to consolidate small parts into larger ones or discard deleted rows.
+
+Materialize lets you specify the ordering it will use to sort these runs of data internally. A well-chosen sort order can unlock optimizations like [filter pushdown](#filter-pushdown), which in turn can make queries and other operations more efficient.
```

Comment:
It seems like the "patterns" section is where we put this sort of doc that covers features that aren't specific to a single type and involve some discussion of implementation details, but let me know if you'd like me to move it!

Reply:
I too am a little confused by what goes into our patterns section, but the pattern of sort order + filter pushdown seems like it would go here.

Reply:
Cool, will leave it alone for now then!